### PR TITLE
[BACKLOG-37966] Toolbar improvements in PUC

### PR DESCRIPTION
**Pull request by AAravind27**
@pentaho/wcag ,Please review

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 15s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **428**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-967/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 15m 16s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **428**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-967/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/Toolbar.java [hunk 1]
```diff
@@ -74,52 +74,56 @@ public Toolbar() {
...
       }
+    });
+  }-*/;
 
-      event.preventDefault();
 
-      if ( nextElement != null ) {
-        nextElement.focus();
+  private boolean itemFilter( Element element ) {
+    boolean result = false;
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/Toolbar.java:106]: Imo, no need for variable.
It's preferable to return the result from inside the if. And return false at the end.

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/Toolbar.java [hunk 2]
```diff
@@ -74,52 +74,56 @@ public Toolbar() {
...
+    super.onAttach();
+
+    jsToolbarController = createJsToolbarController( this, getElement() );
+  }
+
+  @Override
+  protected void onDetach() {
+    if ( jsToolbarController != null ) {
+      destroyJsToolbarController( jsToolbarController );
+    }
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/Toolbar.java:90]: Should null out the jsToolbarController field.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 7s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **428**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-967/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Review by phalguni-a — APPROVED** [review]

**Pull request merged**